Salto-1642: dont separate salesforce installed package custom fields

tamtamirr opened the pull request:
The custom_object_split filter no longer splits custom fields of installed packages to separate files

---
_Release Notes_: 
Salesforce Adapter:
- Installed Packages Custom fields will sit with the rest of the object's CustomFields.
---
_User Notifications_: 
- All of the {OBJECT_NAME}CustomFields.nacl files under the  InstalledPackages directory will be removed from your workspace.
- Your {OBJECT_NAME}CustomFields.nacl files under the Objects directory may change to include the installed packages custom fields. (in case you had CustomFields from installed packages)

tamtamirr commented:
You can see my workspace working on the cpq1 sandbox [here](https://github.com/tamtamirr/remote/tree/cpq-test). I've added two commits, the first one is before my changes, so you could compare.

I've noticed the following new behaviour:
- After my change, it creates `{OBJECT_NAME}CustomFields.nacl` empty file even when theres no custom fields (However, thats the same behaviour as it was before for the Annotations and StandardFields file)

@adamro Wdyt?

adamro commented:
@tamtamirr I didn't do a full review yet, but the regression doesn't seem sensible to me. We have a lot of CustomObjects with no CustomFields (the Standard ones) so it's just a lot of noise.

tamtamirr commented:
@adamro Okay, Added a feature that filters out any empty objects. (Object is empty when it has no annotations and fields). Note - This will also cause deletion of existing empty Annotations and StandardFields files.

adamro commented:
@tamtamirr 
Did we filter out empty CustomFields files specifically before this PR? (if so, why are we not doing that again with the current implementation?)
At what level are you going to filter out empty objects? It sounds like a change that is our of the scope of this PR.

tamtamirr commented:
@adamro
 After discussing with @ori-moisis we decided it's okay okay to filter any empty objects from the result of the split. This won't delete previous empty .nacls. The filter happens as part of the `onFetch`, as follows:
```typescript
const isNotEmptyObject = (customObject: ObjectType): boolean =>
      !(_.isEmpty(customObject.annotations) && _.isEmpty(customObject.fields))
    elements.push(...newSplitCustomObjects.filter(isNotEmptyObject))
```

tamtamirr commented:
WIP
----

After problems I've caused in previous PRs, I've decided to make the changes in the tests as minimal as possible, so I'll ping you both @ori-moisis @adamro when the tests are ready for CR. In the meantime, feel free to comment on the rest :)

tamtamirr commented on packages/salesforce-adapter/test/filters/custom_object_split.test.ts:
```diff
@@ -71,100 +74,127 @@ describe('Custom Object Split filter', () => {
-      expect(Object.values(annotationsObj.fields).length).toEqual(0)
-      expect(annotationsObj.annotations[METADATA_TYPE]).toEqual(CUSTOM_OBJECT)
-    })
+  describe.each`
```
Please refer to the following [doc](https://jestjs.io/docs/api#2-describeeachtablename-fn-timeout) for info

tamtamirr commented:
@ori-moisis @adamro 

Ready for CR :)

Did the minimal changes I could. I'd recommend looking at the commits separately, since the last commit includes the new tests I added which are unrelated to the previous tests.

Note that I've deleted 4 previous tests that were specific to the previous split, those were removed and instead we run the same tests as before for both non-namespace and namespace objects.

Thanks!

ori-moisis reviewed (commented):
discussed f2f - we do not want to move all custom objects out of "InstalledPackages".

tamtamirr commented:
Did many unneeded changes, will close this PR and open a cleaner one.

*CLOSED*
